### DISCUSSION TITLE
Como export meu banco MySQL em uma API Rest apenas de consulta da maneira mais rápida e fácil?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1644] [Perguntas e Respostas]
Galera, 
tenho uma base legal de dados em um MySQL, quero apenas expor ela por uma REST, as tabelas são eventos, organizadores de eventos, generos e etc. 

Quero expor para ter uma api minha de testes enquanto prático e estudo, por isso queria algo simples, sei como fazer tranquilo com php + lumen, ou com o node + express ou no .net/core, mas em todas aqui eu faria o endpoint, conectaria com o orm processando alguns filtros como páginas, busca e etc, e retornaria o json.

Mas gsotaria de saber se vocês conhecem alguma lib, framework ou ferramenta em que eu apenas instale a lib coloque os acessos da base, faça algum filtro para expor apenas as tabelas públicas e dê um start, e a lib/framework/ferramenta faça a magia acontecer por traś.

Conhecem algum?

**Comment by alexedtionweb** [comment]
Crie un usuário somente leitura para o database e realize a instalação da xmysql, uma lib em JS que cria endpoints automáticos.
Vai precisar ter o node instalado.
Segue artigo com explicação:

https://medium.com/@o1lab/build-rest-apis-at-the-speed-of-your-thought-nodejs-mysql-xmysql-95a5eccbd450

**Comment by wilsonneto-swe** [comment]
Shoow @alexedtionweb, muito obrigado!! Exatamente o que precisava, queria só consultas, essa lib superou as expectativas com a possibilidade de gravar dados também.

Forte abraço!

**Comment by eliseumds** [comment]
É vivendo e aprendendo. Eu já estava pensando em coisas mirabolantes, em arquitetura, segurança, etc etc, e às vezes nós precisamos somente de algo rápido e prático como *xmysql*. Participar desse fórum me traz equilíbrio. Obrigado :)
